workers/unity/Assets/Fps/Scripts/GameLogic/Session/SessionObserver.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Improbable.Gdk.Subscriptions;
+using Improbable.Gdk.Session;
+using UnityEngine;
+
+namespace Fps
+{
+    public class SessionObserver : MonoBehaviour
+    {
+        [Require] private SessionWriter sessionWriter;
+        [Require] private TimerWriter timerWriter;
+
+        private const float LobbyTime = 60f;
+        private const float CooldownTime = 60f;
+
+        private float timer;
+
+        private float maxSessionTime;
+        private float currentSessionTime;
+        private float lastUpdate;
+
+        private void OnEnable()
+        {
+            maxSessionTime = timerWriter.Data.MaxTimeSeconds;
+        }
+
+        private void Update()
+        {
+            if (timer > (maxSessionTime + CooldownTime + LobbyTime) && sessionWriter.Data.Status == Status.STOPPING)
```

Comment:
I'd flip the order in which we check, its easier to read that the timer comparison depends on the state.